build/builder.go

```diff
@@ -337,12 +345,18 @@ func (o *Options) shardName(n int) string {
 }
 
 func (o *Options) shardNameVersion(version, n int) string {
-	abs := url.QueryEscape(o.RepositoryDescription.Name)
-	if len(abs) > 200 {
-		abs = abs[:200] + hashString(abs)[:8]
+	var prefix string
+	if o.ShardPrefix != "" {
+		prefix = o.ShardPrefix
+	} else {
+		prefix = url.QueryEscape(o.RepositoryDescription.Name)
+	}
```

Comment:
we should still use url.QueryEscape to protect bad names being used? Or is this intentional to try get things in different directories/etc? Or do we just trust the input? Either way here is a suggestion using new stuff in go to make it a one liner :o
```suggestion
	prefix := url.QueryEscape(cmp.Or(o.ShardPrefix, o.RepositoryDescription.Name))
```